examples/mlee.py

```diff
@@ -0,0 +1,610 @@
+            citation=_CITATION,
+        )
+
+    def _split_generators(
+            self, dl_manager: datasets.DownloadManager
+    ) -> List[datasets.SplitGenerator]:
+        """
+        Step 5: Download and extract the dataset.
+
+        For each config name, run `download_and_extract` from the dl_manager; this will download and unzip any files within a cache directory, specified by `data_dir`.
+
+        `download_and_extract` can accept an iterable object and return the same structure with the url replaced with the path to local files:
+
+        ex:
+        output = dl_manager.download_and_extract({"data1:" "url1", "data2": "url2"})
+
+        output
+        >> {"data1": "path1", "data2": "path2"}
+
+        Nested zip files can be cached also, but make sure to save their path.
+
+        Fill the arguments of "SplitGenerator" with `name` and `gen_kwargs`.
+
+        Note:
+
+        - `name` can be: datasets.Split.<TRAIN/TEST/VALIDATION> or a string
+        - all keys in `gen_kwargs` can be passed to `_generate_examples()`. If your dataset has multiple files, you can make a separate key for each file, as shown below:
+
+        """
+
+        my_urls = _URLs[_DATASETNAME]
+        data_dir = Path(dl_manager.download_and_extract(my_urls))
+        data_files = {
+            "train": data_dir
+                     / "MLEE-1.0.2-rev1"
+                     / "standoff"
+                     / "development"
+                     / "train",
+            "dev": data_dir / "MLEE-1.0.2-rev1" / "standoff" / "development" / "test",
+            "test": data_dir / "MLEE-1.0.2-rev1" / "standoff" / "test" / "test",
+        }
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"data_files": data_files["train"]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"data_files": data_files["dev"]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"data_files": data_files["test"]},
+            ),
+        ]
+
+    def _generate_examples(self, data_files: Path):
+        """
+        Step 6: Create a generator that yields (key, example) of the dataset of interest.
+
+        The arguments to this function come from `gen_kwargs` returned in `_split_generators()`
+
+        The goal of this function is to perform operations on any of the keys of `gen_kwargs` that allow you to extract and process the data.
+
+        The following skeleton does the following:
+
+        - "extracts" abstracts
+        - "extracts" entities, assuming the output is of the form specified in `_info`
+        - "extracts" relations, assuming similarly the output in the form specified in `_info`.
+
+        An assumption in this pseudo code is that the abstract, entity, and relation file all have linking keys.
```

Comment:
Here too, I think the boilerplate here is actively misleading for a participant.

Reply:
Resolved via 6525b94fea387d4fc0346e90f056c507313a393a